tools/run-converter.sh

```diff
@@ -21,11 +21,7 @@ dotnet run --project $CONVERTER_PROJECT -- \
   $SPEC_DIRECTORY/*.md $TEMPLATE \
   -o $OUTPUT_DIRECTORY/standard.docx
 
-if [ "$?" -eq "0" ]
+if [ -n "$GITHUB_OUTPUT" ] && [ "$?" -eq "0" ]
```

Comment:
I don't think we'll ever see a non zero exit code due to set -e. But if we ever did, then when not running on GitHub this would fail because it would go into the else.

If I'm not making sense, I can raise a PR tomorrow - I'm currently only on a mobile and shouldn't be working...

Reply:
Thanks Jon, or just add a suggestion when you have the time. I think we're just talking past each other and I'm missing your point.

Reply:
OK, did a bit of reading. I think we should do one of two things. I prefer the former, but you do more shell programming than I do:

1. instead of `set -e`, use `trap ReportError ERR` instead. Then, the `ReportError` function can write "failed" to GITHUB_OUTPUT. I like this one better because our actions will reliably stop on the first error.
2. Remove the second check, as you point out. It will never be hit. This works *because* the YML files for our actions check for a success code, rather than a failure code. It feels a bit more brittle, but not by much.

What do you think?

Reply:
I prefer option 2, personally. I *believe* that GitHub actions will fail (and immediately stop) automatically if they run a command that fails - as if we'd written "failed" to GITHUB_OUTPUT.

To put it another way: I've never had to explicitly write "failed" to GITHUB_OUTPUT in any other context, and I've seen plenty of actions fail and stop appropriately :)

Option 2 is basically more general IMO - it means our scripts just work in a "normal" way without being aware of GitHub at all.